### ISSUE TITLE
Loading Solution Hangs

### ISSUE THREAD

**Issue by markbjerke** [Microsoft/nodejstools#1457]
##### Actual Behavior 

- NTVS Version:
- Visual Studio Community 2015
- Node.js Version: 
NodeJS - LTS 6.4.9

##### Steps to Reproduce
1. We have problems trying to open our NodeJS console app solutions.  VSS just hangs indefinitely.
2. It looks to me like the analysis that is done on the node_modules folder is not working, and the code
is throwing exceptions and looping forever.  Out team has wasted considerable time trying to use VSS 
for NodeJS development.  
3. The hour glass is up all the time blocking work from being done.  What ever tasks are running in the 
background they block.  That is blocking I/O when you do the check with NPM on my node_modules 
folder.
4. Perhaps the blocking I/O is with the Git repo we are using.  Regardless the fact that VSS regularly hangs
suggests a poor design and poor execution of a poor design.  It's a two for one, of bad software.

**Comment by billti** [comment]
Sorry, to clarify - when you say "VSS", are you referring to "Visual Source Safe" (the version control software)? As you also mention Git later on. (Or do you just mean "VS"?).

We recently fixed a deadlock in the bulid scenario (https://github.com/Microsoft/nodejstools/commit/a9bc9b4cf1d4f5844cbd8aa0004cfdf9445f2141), but I'm unable to repro issues just opening a solution. Do you have any anything specific about your environment that might help us narrow down further (e.g. other extensions installed, etc.)

Do you see issues elsewhere in VS, or is it only this scenario giving you problems? Is this happening for all NTVS projects, new & existing, or just this one existing one? (If the latter, would you be able to provide it for debugging?)

Thanks.

**Comment by billti** [comment]
Also, what NTVS version? I'm not seeing this when using the current pre-releases (https://github.com/Microsoft/nodejstools/releases).

**Comment by kelias** [comment]
I was having the same issue issue in Visual Studio 2017 15.1 (26403.0) using NTVS 1.3 for 2017 (one that came with the product)

It would hang at 88% and never complete. Worked fine if I deleted the node_modules folder.

The problem was that I didn't have the C++ tools installed. Once I did that the project loaded fine. 

This was a basic project with only Express installed.

Might be worth checking if this resolves your issue.
